src/libPMacc/include/mpi/SeedPerRank.hpp

```diff
@@ -55,8 +56,18 @@ namespace mpi
         {
             PMACC_AUTO(&gc, PMacc::Environment<T_DIM>::get().GridController());
 
-            seed ^= gc.getGlobalSize( ) * localShift +
-                    gc.getGlobalRank( );
+            uint32_t rank = gc.getGlobalRank( );
+            /* We put the rank into the upper bits to allow values which start
+             * from zero (e.g. cellIdxs) to be used as additional seed contributors
+             * Those would then write to the lower bits leaving the upper bits alone
+             * which still results in globally unique seeds
+             */
+            uint32_t globalUniqueSeed = reverseBits(rank);
```

Comment:
Really? I thought we agreed on just reversing the rank and xor the localShift. If localShift is a globally constant value of whatever size it doesn't matter though if it is also reversed. Not reversing it and assuming localshift to be small we'd get some more bits set. Only problem would be, if localshift is also counted (e.g. 1 increasing value per species) then XORing this before the reversing would be better.